policy-controller/k8s/status/src/index.rs

```diff
@@ -0,0 +1,222 @@
+                    k8s::Condition {
+                        last_transition_time: k8s::Time(Utc::now()),
+                        message: "".to_string(),
+                        observed_generation: None,
+                        reason: "Accepted".to_string(),
+                        status: "True".to_string(),
+                        type_: "Accepted".to_string(),
+                    }
+                } else {
+                    k8s::Condition {
+                        last_transition_time: k8s::Time(Utc::now()),
+                        message: "".to_string(),
+                        observed_generation: None,
+                        reason: "NoMatchingParent".to_string(),
+                        status: "False".to_string(),
+                        type_: "Accepted".to_string(),
+                    }
+                };
+                gateway::RouteParentStatus {
+                    parent_ref: gateway::ParentReference {
+                        group: Some(POLICY_API_GROUP.to_string()),
+                        kind: Some("Server".to_string()),
+                        namespace: Some(parent_reference_id.namespace.clone()),
+                        name: parent_reference_id.name.clone(),
+                        section_name: None,
+                        port: None,
+                    },
+                    controller_name: STATUS_CONTROLLER_NAME.to_string(),
+                    conditions: vec![condition],
+                }
+            })
+            .collect();
+        let status = gateway::HttpRouteStatus {
+            inner: gateway::RouteStatus {
+                parents: parent_statuses,
+            },
+        };
+        let patch_value = serde_json::json!({
+            "apiVersion": POLICY_API_VERSION,
+            "kind": "HTTPRoute",
+            "name": id.name,
+            "status": status,
+        });
+        k8s::Patch::Merge(patch_value)
+    }
+
+    fn apply_server_update(&self) {
+        for (id, binding) in self.http_routes.iter() {
+            let patch = self.make_http_route_patch(id, binding);
+            if let Err(error) = self.updates.send(Update {
+                id: id.clone(),
+                patch,
+            }) {
+                tracing::error!(%id.namespace, %id.name, %error, "Failed to send HTTPRoute patch")
+            }
+        }
+    }
+}
+
+impl kubert::index::IndexNamespacedResource<k8s::policy::HttpRoute> for Index {
+    fn apply(&mut self, resource: k8s::policy::HttpRoute) {
+        let namespace = resource
+            .namespace()
+            .expect("HTTPRoute must have a namespace");
+        let name = resource.name_unchecked();
+        let id = ResourceId::new(namespace.clone(), name.clone());
+
+        // Create the route binding and insert it into the index. If the
+        // HTTPRoute is already in the index and it hasn't changed, skip
+        // creating a patch.
+        let binding = match RouteBinding::try_from(resource) {
```

Comment:
For each parent in these bindings, do we need to check `self.servers` to see if that Server actually exists?

Reply:
Nope checking if a Server exists is part of `fn make_http_route_patch`. Here we're just collecting the parent references for the actual HTTPRoute spec.